Sucuri/sucuri-wordpress-plugin
# Add Cypress test suite

This PR adds two things:
- Cypress test suite covering about 80% of our features.
- GitHub Actions integration to run the test suite automatically on push or pull request events.

Besides that, I'm also adding [wp-env](https://make.wordpress.org/core/2020/03/03/wp-env-simple-local-environments-for-wordpress/) for easy local development.

To test this PR, start Docker and run:

```
npm install
npx wp-env start
npx cypress run
```

You can also see the _actions_ firing up here: https://github.com/imgerson/sucuri-wordpress-plugin/actions

TODO: We're missing testing for the Firewall tab, but I'll cover that in another PR, as this one is growing fast, and we need to cut the line somewhere. 🎉
